### DISCUSSION TITLE
Request hook in plugin

### DISCUSSION THREAD

**Discussion by tyrunasj** [nitrojs/nitro#1782] [General]
Hi, 
I just wonder when this happens
` nitroApp.hooks.hook("request", (event) => {});`
Before middleware or after when a request hits an endpoint?

**Comment by Holi0317** [comment]
I guess hooks run before middleware. Here is the hook handling section in h3: https://github.com/unjs/h3/blob/8345c1f493abc284e6272621929ecb6385f3e4f2/src/app.ts#L126-L129

Middleware is handled in the `stack`  which will get looped through after hook handling code. I don't understand the code totally but throwing some breakpoints in the server hinted me that conclusion.
